### PR TITLE
expression: implement vectorized evaluation for `builtinWeekOfYearSig` 

### PR DESCRIPTION
**Pull request by yjh126yjh**
### What problem does this PR solve? 
implement vectorized evaluation for builtinWeekOfYearSig #12103 

### What is changed and how it works?

```
goos: windows
goarch: amd64
pkg: github.com/pingcap/tidb/expression
BenchmarkVectorizedBuiltinTimeFunc/builtinWeekOfYearSig-VecBuiltinFunc-4                   26856             44341 ns/op               0 B/op          0 allocs/op
BenchmarkVectorizedBuiltinTimeFunc/builtinWeekOfYearSig-NonVecBuiltinFunc-4                15364             77637 ns/op               0 B/op          0 allocs/op
```
### Check List

Tests

 - Unit test

### CODE REVIEW

**Comment by francis0407** [comment]
Hi @yjh126yjh, thanks for your contribution. 

Please sigh the CLA and fix the format error in the file you have changed.

File: expression/builtin_time_vec.go [hunk 1]
```diff
@@ -699,11 +699,40 @@ func (b *builtinMakeDateSig) vecEvalTime(input *chunk.Chunk, result *chunk.Colum
...
 
 func (b *builtinWeekOfYearSig) vecEvalInt(input *chunk.Chunk, result *chunk.Column) error {
-	return errors.Errorf("not implemented")
+	n := input.NumRows()
+	buf, err := b.bufAllocator.get(types.ETDatetime, n)
+	if err != nil {
+		return err
+	}
+	defer b.bufAllocator.put(buf)
+	if err = b.args[0].VecEvalDuration(b.ctx, input, buf); err != nil {
```
Comment (wshwsh12) [inline expression/builtin_time_vec.go]: s/ VecEvalDuration /VecEvalTime ?

Comment (yjh126yjh) [reply expression/builtin_time_vec.go]: I'll fix that.

**Pull request merged**